pre-push.sh

```diff
@@ -9,15 +9,19 @@ if [ -e "$HOME/.nvm/nvm.sh" ]; then
 fi
 
 # stash un-staged changes
+NUM_STASHES_BEFORE=$(git stash list | wc -l)
```

Comment:
not a huge deal, but I would compare the refs rather than just counts (i.e. with `git rev-parse -q --verify refs/stash`)